src/components/base/BaseHeading.vue

```diff
@@ -0,0 +1,53 @@
+<script>
+import { defineComponent } from 'vue';
+
+export default defineComponent({
+  props: {
+    headingSize: {
+      type: String,
+      default: 'large',
```

Comment:
Since the prop is required I think it doesn't make sense to have a default value. Also valid would make it not required and have a default value instead :)